spec/bolt/node/winrm_spec.rb

```diff
@@ -279,6 +279,20 @@ def stub_winrm_to_raise(klass, message)
     end
   end
 
+  it "passes arguments to the task in the powershell way", winrm: true do
+    contents = <<-PS
+      param ($Name, $Age)
```

Comment:
Can we make this param block typed?

```powershell
[CmdletBinding()]
param (
  [String]
  $Name,
  [Int32]
  $Age
)
```

Reply:
ok

Reply:
OTH, this will mask the fact that the powershell input method provides the values of the correct types even without the need to specify the types in the `param ()` section.
When the types are specified in the `param ()` section the values will be coerced to the correct types irrespective of types of the values in the hash provided by the powershell input method.
So I'd rather keep them untyped.

Reply:
Yes, PowerShell will do a lot of automaticion coercion - you can see some of the details in the gist I put together (which was the basis for the tasks shim in PE) - https://gist.github.com/Iristyle/3b37ad44c93e3e00502d799344294d93

I can understand why'd you want to make sure JSON typed values (i.e. string, number, array, bool, null) all flow through to untyped params on the PS side as you suggest.

I can also understand where you'd want to make sure the coercion behaves correctly for typed params on the PS side (i.e. to make sure JSON types properly coerce to their expected corresponding PS types) - if for no other reason than to document in tests what coercions are allowed. The norm for PowerShell scripts for end users will have the `[Type]` attributes for all parameters from what I've seen - parameter typing is pretty idiomatic for PS.

Reply:
I have added the types to the other two test, it makes a difference here:
https://github.com/puppetlabs/bolt/pull/179/files#diff-a0e64205eda75ed9e253f283c7f02df6R336
Without the typed parameter, the expected string was:
```
foo:  (String), bar: 30 (String)\r\n
```

Reply:
I've updated the first test to use a regexp to match the output to account for the possible differences in the data types (`Int32` vs. `Int64`, `Decimal` vs. `Double`).